src/main/java/mate/academy/bookstorev2/security/CustomUserDetailService.java

```diff
@@ -0,0 +1,26 @@
+package mate.academy.bookstorev2.security;
+
+import lombok.RequiredArgsConstructor;
+import mate.academy.bookstorev2.model.User;
+import mate.academy.bookstorev2.repository.user.UserRepository;
+import org.hibernate.Hibernate;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class CustomUserDetailService implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    @Transactional
+    public UserDetails loadUserByUsername(String email) throws UsernameNotFoundException {
+        User user = userRepository.findUserByEmail(email)
+                .orElseThrow(() -> new UsernameNotFoundException("Can't find user by email"));
+        Hibernate.initialize(user.getRoles());
+        return user;
```

Comment:
```suggestion
        return userRepository.findUserByEmail(email)
                .orElseThrow(() -> new EntityNotFoundException("Can't find user by email"));
```

Reply:
Without @Transactional and Hibernate.initialize(user.getRoles()); there is a Lazy initialization Exception because of Roles